### PR TITLE
feat(bigquery): Custom message when Service Account doesnt have the correct Roles and Permissions

### PR DESCRIPTION
**Pull request by Antonio-RiveroMartnez**
### SUMMARY
`BigQuery` is now returning the project name twice in the error message when connecting, which breaks the Regex in the `custom_errors` dictionary, so we need to adjust it in order to stop displaying the general connection failed message. Also, we are updating the message displayed to users and the tests.

Additionally, in order to actually being able to catch the right error, we are raising the exceptions with their original message instead of the default one, otherwise the exception handler will ignore it and display the default message.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
error

After:
test

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [X] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by AAfghahi — APPROVED** [review]

File: superset/databases/commands/create.py [hunk 1]
```diff
@@ -46,6 +48,34 @@ def run(self) -> Model:
...
+        except DatabaseTestConnectionFailedError as ex:
+            event_logger.log_with_context(
+                action=f"db_creation_failed.{ex.__class__.__name__}",
+                engine=self._properties.get("sqlalchemy_uri", "").split(":")[0],
+            )
+            # if the exception is DatabaseTestConnectionFailedError it has errors in it
+            # so we try to return the message from the errors instead of using the
+            # default message
+            error_list: List[SupersetError] = ex.errors
+            for error in error_list:
```
Comment (hughhhh) [inline superset/databases/commands/create.py:60]: instead of doing a loop can we just do set check?

`if SupersetErrorType.CONNECTION_DATABASE_PERMISSIONS_ERROR in error_list`

Comment (Antonio-RiveroMartnez) [reply superset/databases/commands/create.py:60]: Hmm `error_list` contains a list of `SupersetError`, while `SupersetErrorType.CONNECTION_DATABASE_PERMISSIONS_ERROR` is just a enum, that comparison won't work. Reducing error_list would also require iterating over the list, so, let me see if there's a more "Pythonic" way of checking without looping.

File: superset/databases/commands/create.py [hunk 2]
```diff
@@ -46,6 +48,34 @@ def run(self) -> Model:
...
+                # for all cases.
+                # Why Adding the message for this error type?
+                # Because it's the type returned for Service Accounts without roles
+                # or permissions when connecting to BigQuery DBs, which is
+                # the change we are introducing at this moment.
+                if (
+                    error.error_type
+                    == SupersetErrorType.CONNECTION_DATABASE_PERMISSIONS_ERROR
+                ):
+                    raise DatabaseConnectionFailedError(
```
Comment (hughhhh) [inline superset/databases/commands/create.py:75]: can we just raise the original `DatabaseTestConnectionFailedError`

Comment (Antonio-RiveroMartnez) [reply superset/databases/commands/create.py]: Updated. I am catching the new exception type in the API now along with sending back to the user a dict with the error messages instead of just raw strings.

Comment (hughhhh) [reply superset/databases/commands/create.py]: can you show me the diff between the payload coming back from the API when you send `DatabaseConnectionFailedError` vs the original exception?

Comment (Antonio-RiveroMartnez) [reply superset/databases/commands/create.py]: Before I was raising `DatabaseConnectionFailedError` which only has one error in it, so the API response would look like:
`{message: '<message_here>'}` while now I'm just throwing the original exception (`DatabaseTestConnectionFailed`) which has a list of errors, so the API would respond with a dict like: `{[error_type]: [error_message], [error_type_2]: [error_message_2], ...}` which is already a known pattern to the `DatabaseModal` for error displaying.

So basically after the change based on the input you gave me, I'm doing the latter.

For example:

Before when throwing a `DatabaseConnectionFailedError` the API would respond:

**Review by jinghua-qa — APPROVED** [review]
Able to see the custom msg for error msg. LGTM

**Review by hughhhh — APPROVED** [review]

File: superset/databases/commands/test_connection.py [hunk 3]
```diff
@@ -145,7 +152,7 @@ def ping(engine: Engine) -> bool:
             )
             # check for custom errors (wrong username, wrong password, etc)
             errors = database.db_engine_spec.extract_errors(ex, context)
-            raise DatabaseTestConnectionFailedError(errors) from ex
+            raise SupersetErrorsException(errors) from ex
```
Comment (eschutho) [inline superset/databases/commands/test_connection.py:155]: @Antonio-RiveroMartnez I'm looking into some 500 errors for test connections, and I saw this change. What would happen if I were to change this back to raise the `DatabaseTestConnectionFailedError` which is a 422? Would it break anything from this PR? I checked through the implementation and it looks like it should still catch the SupersetErrorsExceptions because it is the base class, but I wanted to see if I might be missing anything here.

Comment (Antonio-RiveroMartnez) [reply superset/databases/commands/test_connection.py:155]: Hey @eschutho, the problem of doing that is that we would lost the `original message` and instead the message from DatabaseTestConnectionFailedError would be used (`Connection failed, please check your connection settings`) and not the custom message with roles after parsing the original message.
